models/src/node/mod.rs

```diff
@@ -148,6 +156,25 @@ impl BottlerocketShadow {
             node_status.current_state == self.spec.state
         })
     }
+
+    /// Returns whether or not a node has crashed.
+    pub fn has_crashed(&self) -> bool {
+        self.status.as_ref().map_or(false, |node_status| {
+            node_status.current_state == BottlerocketShadowState::ErrorReset
+        })
+    }
+
+    /// Order BottleRocketShadow based on crash_count in status
+    /// to determine the priority to be handled by the controller.
+    /// Uninitialized status should be considered as lowest priority.
+    pub fn compare_status(&self, other: &Self) -> Ordering {
```

Comment:
Naming suggestion:

```suggestion
    pub fn compare_crash_count(&self, other: &Self) -> Ordering {
```

I really like the use of `cmp` here, by the way :).

Reply:
It's tricky to implement `cmp` for `BottlerocketShadow` though, that's why I did this method.